### PR TITLE
remove darwin arm64 from packaging

### PR DESCRIPTION
**Pull request by sharbuz**
## What is the problem this PR solves?

Jenkins->Buildkite pipelines migration 

Fix errors:
https://buildkite.com/elastic/auditbeat/builds/1848#018dcc3d-c30d-49a6-87af-7dc2643bf087/340-402
https://buildkite.com/elastic/filebeat/builds/3176#018dcc40-20e5-4f4e-ac2f-184be9b07ab2/339-403
https://buildkite.com/elastic/heartbeat/builds/1837#018dcc3e-983e-40ba-8015-292aa9b167f1/340-401

## Related issues

https://github.com/elastic/ingest-dev/issues/1693

### CODE REVIEW

File: .buildkite/filebeat/scripts/package-step.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ if are_files_changed "$changeset"; then
       - label: ":ubuntu: Packaging Linux X86"
         key: "package-linux-x86"
         env:
-          PLATFORMS: "+all linux/amd64 linux/arm64 windows/amd64 darwin/amd64 darwin/arm64"
```
Comment (cmacknz) [inline .buildkite/filebeat/scripts/package-step.sh:21]: We publish a darwin/aarach64 image for these Beats, are we sure removing it is right? 

* https://artifacts.elastic.co/downloads/beats/filebeat/filebeat-8.12.1-darwin-aarch64.tar.gz
* https://artifacts.elastic.co/downloads/beats/auditbeat/auditbeat-8.12.1-darwin-aarch64.tar.gz
* https://artifacts.elastic.co/downloads/beats/heartbeat/heartbeat-8.12.1-darwin-aarch64.tar.gz

Is this right?

Comment (sharbuz) [reply .buildkite/filebeat/scripts/package-step.sh:21]: let me check one more time the root cause of the problem

Comment (sharbuz) [reply .buildkite/filebeat/scripts/package-step.sh:21]: It looks like the darwin/arm64 platform isn't supported by only in 7.17 branch.
@cmacknz Please correct me if I'm wrong.
If it's correct, I'll recreate PR for 7.17 only
All mentioned in the issue description builds are related to 7.17.

Comment (v1v) [reply .buildkite/filebeat/scripts/package-step.sh:21]: That's correct, see the below details:

The current packaging pipeline for `7.17` is not building `arm64`:
- https://github.com/elastic/beats/blob/6b671dd901b6b1824d12d33568859af2a5937402/.ci/packaging.groovy#L306-L325

As far as I see `darwin/arm64` was added to `>= 8.2` as per https://github.com/elastic/beats/pull/29585

https://www.elastic.co/downloads/past-releases/elastic-agent-8-1-3 vs https://www.elastic.co/downloads/past-releases/elastic-agent-8-2-0

and no arm64 artifacts in https://www.elastic.co/downloads/past-releases/elastic-agent-7-17-18

Comment (sharbuz) [reply .buildkite/filebeat/scripts/package-step.sh:21]: I've created a new one https://github.com/elastic/beats/pull/38098

**Pull request closed without merging**